### ISSUE TITLE
The bottom parameter of plt.hist() shifts the data as well, not just the baseline

### ISSUE THREAD

**Issue by torokati44** [matplotlib/matplotlib#16478]
When passing a `bottom` parameter to `plt.hist()`, the entire histogram is shifted by that amount, as a whole, instead of just the baseline, as the documentation would suggest.

See this simple example:

```python
import numpy as np
import matplotlib.pyplot as plt
np.random.seed(42)
x = np.random.randn(10000)
b, e = np.histogram(x)
plt.hist(e[:-1], e, weights=b)
plt.show()
```

This output of this is:
![unshifted](https://user-images.githubusercontent.com/288816/74338584-c72edf80-4da2-11ea-8c0b-ca6c67dd9cbf.png)

This is all fine and good.

When adding the `bottom=1500` parameter:

```python
import numpy as np
import matplotlib.pyplot as plt
np.random.seed(42)
x = np.random.randn(10000)
b, e = np.histogram(x)
plt.hist(e[:-1], e, weights=b, bottom=1500)
plt.show()
```

This is what I get:

![shifted](https://user-images.githubusercontent.com/288816/74338677-f04f7000-4da2-11ea-9dfd-77fde4031882.png)

The "indicated values" of the histogram have shifted up by `1500` as well, not just the baseline. By baseline, I mean the "straight horizontal" outline of the filled region.

What I was expecting instead, can also be achieved by "unshifting" the actual data as well, by subtracting the value of the `bottom` parameter from it:

```python
import numpy as np
import matplotlib.pyplot as plt
np.random.seed(42)
x = np.random.randn(10000)
b, e = np.histogram(x)
plt.hist(e[:-1], e, weights=b-1500, bottom=1500)
plt.show()
```

See:
![corrected](https://user-images.githubusercontent.com/288816/74338682-f47b8d80-4da2-11ea-83dc-080645efed79.png)

It might be the case that this is entirely what the original intent was behind the functionality, but at the very least, the documentation is misleading in my opinion.
Also, this might apply to `plt.bar()` as well.

**Comment by ImportanceOfBeingErnest** [comment]
This is indeed the intended functionality. Please feel free to suggest a better or more accurate phrasing of the documentation.

**Comment by jklymak** [comment]
I agree that this doesn’t do what the docstring says.  If this is truly the intended behaviour what is the point of it?  It seems at the very least the kwarg is misnamed?

**Comment by ImportanceOfBeingErnest** [comment]
The point of it would be to be able to create stacked histograms manually, i.e. 

```
some_data, other_data = np.random.randn(2,500)
h, bins, bars = plt.hist(some_data)
plt.hist(other_data, bins=bins, bottom=h)
```

**Comment by torokati44** [comment]
I think correcting the documentation to say instead of "shifts the baseline" something like "shifts the entire histogram, or individual bars (if it's a scalar or an array, respectively)" would make it less confusing. And then, it could even be renamed to something "shift"y.
Maybe even mentioning that this will change the "indicated value" (so, the top of the rectangles too, not just the bottom) could help. And perhaps adding the same example you just wrote, showing where this behavior could be useful.
Then people who want what my 3rd screenshot shows, can be sure that they will have to shift the data "back down" to get the desired result.

**Comment by jklymak** [comment]
I think something like that would be better.  I am not sure we can easily change the name of the kwarg, but properly documenting it would be great.

**Comment by torokati44** [comment]
Fixed by #16480.
